corehq/motech/fhir/utils.py

```diff
@@ -53,3 +58,20 @@ def update_fhir_resource_property(case_property, fhir_resource_type, fhir_resour
                                                       resource_type=fhir_resource_type)
         fhir_resource_prop.jsonpath = fhir_resource_prop_path
         fhir_resource_prop.save()
+
+
+def build_capability_statement(domain):
+    """
+    Builds the FHIR capability statement including the OAuth URL extensions for SMART
+    https://hl7.org/fhir/smart-app-launch/conformance/index.html
+    """
+    with open(os.path.join(os.path.dirname(__file__), 'data', 'capability_statement.json'), encoding='utf-8') as f:
+        data = (
+            f.read()
+            .replace('__TOKEN_URL__', absolute_reverse("oauth2_provider:token"))
+            .replace('__AUTHORIZE_URL__', absolute_reverse("oauth2_provider:authorize"))
+            .replace('__FHIR_VERSION__', FHIR_VERSION_4_0_1)
+            .replace('__PUBLISHED_DATE__', CAPABILITY_STATEMENT_PUBLISHED_DATE)
+        )
+        statement = json.loads(data)
```

Comment:
Instead of storing this in a file, or caching it in memory, we could just define it as a dictionary, and update it using the `deepmerge()` function that I killed (with profound sadness) in [this commit here](https://github.com/dimagi/commcare-hq/pull/29456/commits/cc279b3393b897c97f676f0559ea880c8085ff47).

e.g.
```python
    statement = deepmerge(statement, {
        'rest': [{'security': {'extension': {'extension': [
            {'valueUri': absolute_reverse("oauth2_provider:token")},
            {'valueUri': absolute_reverse("oauth2_provider:authorize")},
        ]}}}],
        'fhirVersion': FHIR_VERSION_4_0_1,
        'date': CAPABILITY_STATEMENT_PUBLISHED_DATE,
    })
```